cmd/podman/generate/spec.go

```diff
@@ -0,0 +1,71 @@
+package pods
+
+import (
+	"fmt"
+	"io/ioutil"
+
+	"github.com/containers/common/pkg/completion"
+	"github.com/containers/podman/v4/cmd/podman/common"
+	"github.com/containers/podman/v4/cmd/podman/registry"
+	"github.com/containers/podman/v4/cmd/podman/utils"
+	"github.com/containers/podman/v4/pkg/domain/entities"
+	"github.com/spf13/cobra"
+)
+
+var (
+	specCmd = &cobra.Command{
+		Use:               "spec [options] {CONTAINER|POD}",
+		Short:             "Generate Specgen JSON based on containers or pods",
+		Long:              "Generate Specgen JSON based on containers or pods",
+		RunE:              spec,
+		Args:              cobra.ExactArgs(1),
+		ValidArgsFunction: common.AutocompleteContainersAndPods,
+		Example:           `podman generate spec ctrID`,
+	}
+)
+
+var (
+	opts *entities.GenerateSpecOptions
+)
+
+func init() {
```

Comment:
Given this accepts both containers and pods, would flags to force one or the other (`podman generate spec --ctr` or similar) make sense?

Reply:
`podman inspect` has a `--type` flag that I'd prefer for consistency.

But we could add that in a follow-up PR.